### PR TITLE
ENT-1314: Modify the locking and transaction size for the hypervisor report

### PR DESCRIPTION
**Pull request by Januson**
- Split transaction into a series of bulk operations per 10 items

### CODE REVIEW

**Comment by barnabycourt** [comment]
I see where the flush() is called by bulkUpdate() but the transaction is still open. Won't this still be holding locks for all the hypervisors for the duration of the processing?

File: server/src/main/java/org/candlepin/model/ConsumerCurator.java [hunk 1]
```diff
@@ -622,6 +622,34 @@ private void validateFacts(Consumer consumer) {
         }
     }
 
+    /**
+     * @param consumers consumers to create
+     * @return created consumers
+     */
+    @Transactional
+    public Set<Consumer> bulkCreate(Set<Consumer> consumers) {
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/model/ConsumerCurator.java]: These methods already exist as saveAll and deleteAll in AbstractHibernateCurator

Comment (Januson) [reply server/src/main/java/org/candlepin/model/ConsumerCurator.java]: Done.

**Comment by Januson** [comment]
> I see where the flush() is called by bulkUpdate() but the transaction is still open. Won't this still be holding locks for all the hypervisors for the duration of the processing?

I extracted a piece of Job's logic into a separate class so that the job itself doesn't have to be transactional. Since job is not transactional, transaction should end when the bulkUpdate method returns.

File: server/src/main/java/org/candlepin/model/ConsumerCurator.java (hunk 1 above)
Comment (Ceiu) [reply server/src/main/java/org/candlepin/model/ConsumerCurator.java]: :+1:

**Comment by wottop** [comment]
Those spec failures need to be fixed. You are getting 404 errors on get_consumer.

**Comment by Januson** [comment]
> The cause of the spec failures need to be fixed. You are getting 404 errors on get_consumer.

Done.

**Comment by Ceiu** [comment]
@wottop Is this still needing review or are we good to merge with the fixed spec test ?

**Comment by wottop** [comment]
@Ceiu That was my only concern. If you are good, go ahead.

**Pull request merged**